Dart/src/com/jetbrains/lang/dart/ide/findUsages/DartServerFindUsagesHandler.java

```diff
@@ -64,6 +66,19 @@ public boolean processInReadAction(SearchResult result) {
     return true;
   }
 
+  private static PsiElement findPsiElement(PsiElement context, Location location) {
+    final String locationFilePath = location.getFile();
+    final PsiFile psiFile = findPsiFile(context, locationFilePath);
+    if (psiFile != null) {
+      final int offset = location.getOffset();
+      PsiElement result;
+      if ((result = PsiTreeUtil.findElementOfClassAtOffset(psiFile, offset, DartReference.class, false)) != null) {
+        return result;
+      }
+    }
+    return null;
```

Comment:
We loose some usages this way. At least from doc comments where references are not parsed. May be some other cases as well.
May be if we failed to find a DartReference matching region we can take any deepest PsiElement that includes the region.